workers/unity/Packages/io.improbable.gdk.tools/PortForwardingWindow.cs

```diff
@@ -0,0 +1,179 @@
+using System.Threading;
+using System.Threading.Tasks;
+using UnityEditor;
+using UnityEngine;
+
+namespace Improbable.Gdk.Tools
+{
+    internal class PortForwardingWindow : EditorWindow
+    {
+        private string deploymentName = "";
+        private string workerId = "";
+        private int port;
+
+        private Task<RedirectedProcessResult> activePortForwarding;
+        private CancellationTokenSource tokenSrc;
+
+        private bool isConnected;
+        private bool portForwardingHasFailed;
+
+        [MenuItem("SpatialOS/Port Forwarding", isValidateFunction: false, priority: MenuPriorities.PortForwarding)]
+        public static void ShowWindow()
+        {
+            GetWindow<PortForwardingWindow>().Show();
+        }
+
+        private void OnEnable()
+        {
+            titleContent = new GUIContent("Port Forwarding");
+        }
+
+        public void OnGUI()
+        {
+            using (new EditorGUI.DisabledScope(activePortForwarding != null))
+            using (new EditorGUILayout.VerticalScope())
+            {
+                deploymentName = EditorGUILayout.TextField("Deployment Name", deploymentName);
+                workerId = EditorGUILayout.TextField("Worker ID", workerId);
+                port = EditorGUILayout.IntField("Port", port);
+
+                var errors = GetErrors().ToList();
+
+                if (errors.Any())
+                {
+                    EditorGUILayout.HelpBox(
+                        $"Cannot forward port due to the following errors:\n\n{string.Join("\n", errors)}",
+                        MessageType.Error);
+                }
+
+                using (new EditorGUI.DisabledScope(errors.Any()))
+                {
+                    if (GUILayout.Button("Forward port"))
+                    {
+                        portForwardingHasFailed = false;
+                        isConnected = false;
+                        activePortForwarding = RunPortForwarding();
+                    }
+                }
+            }
+
+            var messageType = MessageType.None;
+            string message = null;
+
+            if (isConnected)
+            {
+                messageType = MessageType.Info;
+                message = $"Connected successfully to {workerId}:{port} on {deploymentName}!\n\nAssembly reloading locked.";
+            }
+            else if (activePortForwarding != null)
+            {
+                messageType = MessageType.Warning;
+                message = "Establishing connection...\n\nAssembly reloading locked.";
+            }
+            else if (portForwardingHasFailed)
+            {
+                messageType = MessageType.Error;
+                message = "Failed to establish connection. Check the Unity Console for more information.";
+            }
+
+            if (message != null)
+            {
+                EditorGUILayout.HelpBox(message, messageType);
+            }
+
+            if (activePortForwarding != null)
```

Comment:
Can combine the two if statements

Reply:
True, although I feel like the intent is clearer as written.

I read this as: 

- If the task isn't running, don't render the button.
- If the button is clicked, cancel the task.

Whereas if its combined:

- If the task is running and the button is clicked cancel the task

Thoughts?

Reply:
I don't feel particularly strongly either way!